base/iterators.jl

```diff
@@ -1171,10 +1171,10 @@ function length(itr::PartitionIterator)
     return cld(l, itr.n)
 end
 
-function iterate(itr::PartitionIterator{<:AbstractRange}, state=1)
-    state > length(itr.c) && return nothing
+function iterate(itr::PartitionIterator{<:AbstractVector}, state = firstindex(itr.c))
+    state > lastindex(itr.c) && return nothing
     r = min(state + itr.n - 1, length(itr.c))
-    return @inbounds itr.c[state:r], r + 1
+    return @inbounds view(itr.c, state:r), r + 1
```

Comment:
If this change doesn't make a difference, there's no need to change it.

Reply:
I think the view makes it consistent with the docstring

Reply:
I am a bit hesitant about this, at least as part of this PR. Since this is supposed to just fix a bug can we remove it for now and perhaps propose that change in a separate PR?